### ISSUE TITLE
creating random example for headers and Support for multiple example

### ISSUE THREAD

**Issue by arvindjangra1996** [kiwicom/schemathesis#582]
hi.
There is issue, when creating example for headers.
it is generating random header value ( not always ), and then raising exception (API will not be executed in this case).
Is it possible to just skip adding header example. 
`in _hypothesis.py `
` PARAMETERS = frozenset(("path_parameters", "headers", "cookies", "query", "body", "form_data"))
`

And can I expect support for multiple example in OPENAPI schema in future release. Like executing this API two times : 
1. `www.xyz.com/fetch?limit = 0`
2. `www.xyz.com/fetch?limit = 50`
```
parameters:
  - in: query
    name: limit
    schema:
      type: integer
      maximum: 50
    examples:       # Multiple examples
      zero:         # Distinct name
        value: 0    # Example value
        summary: A sample limit value  # Optional description
      max: # Distinct name
        value: 50   # Example value
        summary: A sample limit value
```

**Comment by Stranger6667** [comment]
Hi @arvindjangra1996 !

1. Could you, please, provide some schema example where the issue will occur?

You can change the examples list with the `before_add_examples` hook (see [docs](https://schemathesis.readthedocs.io/en/stable/customization.html#before-add-examples)) with something like this:

```
@schemathesis.hooks.register
def before_add_examples(context, examples):
    examples[:] = []
```

This hook will remove all examples, but you can find the only one you need and remove it.

2. It is a cool feature, indeed :) I'd split it into a separate issue

**Comment by arvindjangra1996** [comment]
```
parameters:
  - in: header
    name: anyKey
    schema:
      type: integer
```
It is not happening every time. Sometime it runs, sometime it doesn't. I cant figured out why is this happening.

**Comment by Stranger6667** [comment]
Could you try with `required: true` in the parameter definition? Otherwise, it is optional and sometimes it will be `None`

**Comment by arvindjangra1996** [comment]
No, it is optional header.  We cannot set required as true

**Comment by Stranger6667** [comment]
Hm, and what kind of exception is happening in this case?

**Comment by arvindjangra1996** [comment]
this is creating something like this: 
`Falsifying example: {headers={'userkey': '\r\U0004f9e5'}}`

**Comment by arvindjangra1996** [comment]
hey, any update?

**Comment by Stranger6667** [comment]
Hi, sorry, I will get back to you today

**Comment by Stranger6667** [comment]
I have a guess about it, let's check this one and if I am right, then it is a bug in Schemathesis, otherwise, I'd like to ask you some additional details, because I can't reproduce it.

There is `\r` in your header value above, which is not actually valid in headers, but since it is generated somehow, it can come from the Schemathesis strategies or it might be modified via hooks. Do you have custom hooks like `before_generate_headers` where strategy is replaced? Or it came from a default Schemathesis's strategy?

If it is not the case, then I have the following thoughts.

From what I understand at the moment we have the following behavior:
- There is an optional parameter in `header` with type `integer` type;
- Sometimes it is generated and sometimes it is not;
- When it is generated it causes `Falsifying example` exception;

Using the parameter definition you provided I tried to reproduce this behavior with the following minimal schema:

```
{
    "openapi": "3.0.2",
    "info": {"title": "Test", "description": "Test", "version": "0.1.0"},
    "servers": [{"url": "http://127.0.0.1:8081/{basePath}", "variables": {"basePath": {"default": "api"}}}],
    "paths": {
        "/success": {
            "get": {
                "parameters": [
                    {
                        "name": "anyKey",
                        "in": "header",
                        "schema": {"type": "string"},
                    }
                ],
                "responses": {"200": {"description": "OK"}},
            }
        }
    },
}
```

Then `/api/success` endpoint was running on out [built-in test webserver](https://github.com/kiwicom/schemathesis/blob/master/test/apps/_aiohttp/handlers.py#L9)

```
./test_server.sh 8081 --endpoints=success
```

Then I run Schemathesis like this:

```
schemathesis run --hypothesis-max-examples=1000 --show-errors-tracebacks http://0.0.0.0:8081/swagger.yaml
```

And my output was:

```
======================================================= Schemathesis test session starts ======================================================
platform Linux -- Python 3.8.2, schemathesis-1.6.1, hypothesis-5.10.4, hypothesis_jsonschema-0.12.1, jsonschema-3.2.0
rootdir: /schemathesis
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/schemathesis/.hypothesis/examples')
Schema location: http://0.0.0.0:8081/swagger.yaml
Base URL: http://0.0.0.0:8081
Specification version: Open API 3.0.2
Workers: 1
collected endpoints: 1

GET /api/success E                                                                                                                      [100%]

==================================================================== ERRORS ===================================================================
______________________________________________________________ GET: /api/success ______________________________________________________________
Traceback (most recent call last):
  File "/schemathesis/runner/impl/core.py", line 149, in run_test
    test(checks, targets, result, headers=headers, **kwargs)
  File "/schemathesis/src/schemathesis/runner/impl/core.py", line 212, in network_test
    case: Case,
  File "/lib/python3.8/site-packages/hypothesis/core.py", line 1095, in wrapped_test
    raise the_error_hypothesis_found
  File "/schemathesis/src/schemathesis/_hypothesis.py", line 109, in is_valid_header
    if not utils.is_latin_1_encodable(value):
  File "/schemathesis/src/schemathesis/utils.py", line 58, in is_latin_1_encodable
    value.encode("latin-1")
AttributeError: 'int' object has no attribute 'encode'

=================================================================== SUMMARY ===================================================================

No checks were performed.

============================================================== 1 errored in 0.03s =============================================================
```

My environment is:
- Python 3.8.2
- Schemathesis 1.6.1
- hypothesis-5.10.4
- hypothesis_jsonschema-0.12.1
- pytest-5.4.1

WIth the in-code test, I have the same behavior:

```
schema = schemathesis.from_dict(RAW_SCHEMA)  # the same as above

@schema.parametrize()
def test(case):
    case.call(base_url="http://127.0.0.1:8081")
```

The error traceback is expected to be like this since headers should be latin-1 encodable values of type string. There is no implicit coercion from `int` to `str` and I am not sure how to reproduce the behavior you noted with this header definition. After changing `type` to `string` I still can not reproduce it.

Could you, please provide additional details about how you run it? Is it a CLI run or in-code? Did I miss something important in the schema definition?

Python/Schemathesis/hypothesis/hypothesis-jsonschema versions will also be useful.

Does it occur when you run the tests with `phases=[Phase.explicit]` setting? or equivalent `--hypothesis-phases=explicit` CLI option?

**Comment by Stranger6667** [comment]
Oh, ignore my comment above, I found the problem - no validity filters are attached when examples are generated - https://github.com/kiwicom/schemathesis/blob/master/src/schemathesis/_hypothesis.py#L84

I will work on a fix

**Comment by Stranger6667** [comment]
The fix is released in `1.6.2`. Let me know if it fixes the issue on your side.

The multiple examples feature is here - #589. I'd likely work on it after finishing Open API links & the bug with `base_url` in a few weeks. There is also #467 which I'll try to ship together with that one
